src/ngraph/node.hpp

```diff
@@ -268,6 +275,7 @@ namespace ngraph
         std::string m_friendly_name;
         const std::string m_unique_name;
         static std::atomic<size_t> m_next_instance_id;
+        std::unordered_set<std::string> m_provenance_tags;
```

Comment:
would it make sense to make this a set of *const* std::string?  would we ever want to change a provenance tag once it is added?

Reply:
It did sound more robust to me too, however my finding is, it is not a valid C++. I did a local testing in my VS (unordered_set<const string> s;) and it says "error C2338: The C++ Standard forbids containers of const elements because allocator<const T> is ill-formed." I was also not able to get it compiled on the linux server in our lab.